src/System.Private.ServiceModel/tests/Scenarios/Contract/Service/ServiceContractTests.cs

```diff
@@ -705,8 +708,42 @@ public static void BasicHttp_Abort_ChannelFactory_Operations_Active()
             Assert.True(factory.State == CommunicationState.Closed,
                         String.Format("Expected factory state 'Closed', actual was '{0}'", factory.State));
 
-            Assert.Throws<TaskCanceledException>(() => t1.GetAwaiter().GetResult());
-            Assert.Throws<TaskCanceledException>(() => t2.GetAwaiter().GetResult());
+            Exception exception1 = null;
+            Exception exception2 = null;
+            string actualEcho1 = null;
+            string actualEcho2 = null;
+
+            // Verification is slightly more complex for the close with active operations because
+            // we don't know which might have completed first and whether the channel factory
+            // was able to close and dispose either channel before it completed.  So we just
+            // ensure the Tasks complete with an exception or a successful return and have
+            // been closed by the factory.
+            try
+            {
+                actualEcho1 = t1.GetAwaiter().GetResult();
+            }
+            catch (Exception e)
+            {
+                exception1 = e;
+            }
+
+            try
+            {
+                actualEcho2 = t2.GetAwaiter().GetResult();
+            }
+            catch (Exception e)
+            {
+                exception2 = e;
+            }
+
+            Assert.True(exception1 != null || actualEcho1 != null, "First operation should have thrown Exception or returned an echo");
```

Comment:
Under what condition would the expression be `false`?

Reply:
Good question, Ron's code, I assume it wasn't intentional.
Will open a bug to look at this more carefully.
For now won't block this PR as it is a straight port.